lib/GADS.pm

```diff
@@ -3024,6 +3025,89 @@ prefix '/:layout_name' => sub {
 
     };
 
+    any ['get', 'post'] => '/historic_purge/' => require_login sub {
+        my $layout = var('layout') or pass;
+
+        forwardHome({ danger => "You do not have permission to manage deleted records"}, '')
+            unless $layout->user_can("purge");
+
+        my $user        = logged_in_user;
+        my $view        = current_view($user, $layout);
+
+        my $records = GADS::Records->new(
+            user                => $user,
+            layout              => $layout,
+            schema              => schema,
+            include_children    => 1,
+            view                => $view,
+        );
+
+        my $params = +{};
+        my @record_data;
+
+        if(param 'submit') {
+            my @columns = body_parameters->get_all('column_id');
+            $records->columns(\@columns);
+        } elsif(param 'purge') {
+            my @columns = body_parameters->get_all('column_id');
+            $records->columns(\@columns);
+        }
+
+        unless(param 'purge') {
+            while (my $record = $records->single) {
+                foreach my $column (@{$record->columns_render}) {
+                    next if $column->internal;
+
+                    my $clone = $record->clone;
```

Comment:
This is to count the "correct" number of versions, rather than using `$record->versions`

Reply:
Unfortunately I think use `clone()` and `find_chronology_id` are going to make this very slow for large systems and therefore not an option. The latter is already very slow for single records, so doing it across many records will be problematic I think.

Reply:
I'm inclined to agree, in retrospect!